ocaml/xapi/wlb_reports.ml

```diff
@@ -98,129 +98,59 @@ open Stdext.Xstringext
 module D = Debug.Make(struct let name="wlb_reports" end)
 open D
 
-let report_tokens = ("<XmlDataSet>", "</XmlDataSet>")
-let diagnostics_tokens = ("<DiagnosticData>", "</DiagnosticData>")
+let report_tag = "XmlDataSet"
+let diagnostics_tag = "DiagnosticData"
 
 let bufsize = 16384
 
 let hex_entity s =
   (*debug "hex_entity %s" s; *)
   char_of_int (int_of_string ("0" ^ (String.sub s 1 (String.length s - 1))))
 
-let trim_and_send method_name (start_str, end_str) recv_sock send_sock =
+let trim_and_send method_name tag recv_sock send_sock =
   let recv_buf = Buffer.create bufsize in
-  let send_buf = Buffer.create bufsize in
-  let recv_state = ref 1 in
-  let send_state = ref 1 in
-  let entity = ref "" in
-
   let fill () =
```

Comment:
I would like to suggest the following commit message:

```
This commit refactors the parsing of WLB XML messages. 

* It avoids a problem with the previous disign: XML was read in blocks and parsing it could fail at block boundaries
* Parsing of XML is now delegated to an XML parser.
* The "send" function no longer escapes special characters and emits the message as is
```